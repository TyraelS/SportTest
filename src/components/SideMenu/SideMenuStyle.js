import styled from 'styled-components';

const SideMenuStyle = styled.div`
  width: 25%;
  height: 100vh;
  display: flex;
  flex-direction: column;
  overflow-y: scroll;
`;

export default SideMenuStyle;
