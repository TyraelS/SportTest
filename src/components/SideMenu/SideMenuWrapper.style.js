import styled from 'styled-components';

const SideMenuWrapper = styled.div`
  width: 300px;
  min-height: 400px;
  display: flex;
  flex-direction: column;
`;

export default SideMenuWrapper;
