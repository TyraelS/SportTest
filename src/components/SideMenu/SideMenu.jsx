import React from 'react';
import SideMenuStyle from './SideMenuStyle';
import SideMenuItem from '../SideMenuItem/';

const displayName = 'SideMenu';

export default function SideMenu({ sports, handleShowEvents }) {
  console.log('Current props:', sports.toJS());
  return (
    <SideMenuStyle>
      render time = {new Date().toLocaleTimeString()}
      {sports
        ? sports.map(item => (
            <SideMenuItem key={item.id} id={item.id} onClick={handleShowEvents}>
              {item.name}
            </SideMenuItem>
          ))
        : 'Smth went wrong'}
    </SideMenuStyle>
  );
}

SideMenu.displayName = displayName;
