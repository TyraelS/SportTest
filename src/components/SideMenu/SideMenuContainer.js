import {
  compose,
  lifecycle,
  setDisplayName,
  mapProps,
  withHandlers,
  withState,
  pure
} from 'recompose';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import SideMenu from './SideMenu';
import { fetchSports } from '../../actions/fetchSports';
import { fetchLeagues } from '../../reducers/leagues';
import generate from '../../utils/generateTimestamp';
import {
  setSportsTimestamp,
  setLeaguesTimestamp
} from '../../reducers/responses';
import getSportsWithCounters from '../../selectors/getSports';

export const mapStateToProps = state => {
  return {
    sports: getSportsWithCounters(state)
  };
};

export const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      dispatchFetchSports: fetchSports,
      fetchLeagues: fetchLeagues,
      setSportsTimestamp: setSportsTimestamp,
      setLeaguesTimestamp: setLeaguesTimestamp
    },
    dispatch
  );

export const fetchData = function() {
  const {
    dispatchFetchSports,
    currentSportId,
    fetchLeagues,
    setSportsTimestamp,
    setLeaguesTimestamp
  } = this.props;
  const timestamp = generate();
  setSportsTimestamp(timestamp);
  setLeaguesTimestamp(timestamp);
  dispatchFetchSports(timestamp);
  currentSportId && fetchLeagues(currentSportId, timestamp);
};

export const fetchOnMount = {
  componentDidMount() {
    const fetchDataBind = fetchData.bind(this);
    fetchDataBind();
    setInterval(() => {
      fetchDataBind();
    }, 30000);
  }
};

export const mapContainerProps = props => {
  return {
    ...props,
    sports: props.sports
  };
};

export const handleShowEvents = {
  handleShowEvents: ({
    setCurrentSportId,
    fetchLeagues,
    setLeaguesTimestamp,
    currentSportId
  }) => event => {
    setCurrentSportId(event.target.id);
    const timestamp = generate();
    setLeaguesTimestamp(timestamp);
    fetchLeagues(event.target.id, timestamp);
  }
};

export const enhance = compose(
  setDisplayName('SideMenuContainer'),
  connect(
    mapStateToProps,
    mapDispatchToProps
  ),
  withState('currentSportId', 'setCurrentSportId', null),
  lifecycle(fetchOnMount),
  mapProps(mapContainerProps),
  withHandlers(handleShowEvents),
  pure
);

export default enhance(SideMenu);
