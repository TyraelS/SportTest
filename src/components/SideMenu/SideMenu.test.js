import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';

import SideMenu from './SideMenu';

describe('Given the SideMenu component', () => {
  let sports = fromJS({
    SBTC_01: {
      id: 'SBTC_01',
      name: 'SortofLeague'
    },
    SBTC_02: {
      id: 'SBTC_02',
      name: 'AnotherLeague'
    }
  });
  const sportItemClick = jest.fn();
  const currentSportId = 'SBTC_01';
  describe('when the component is rendered', () => {
    const component = shallow(
      <SideMenu
        sports={sports}
        sportItemClick={sportItemClick}
        currentSportId={currentSportId}
      />
    );
    it('should match the snapshot 1', () => {
      expect(component).toMatchSnapshot();
    });
  });
  describe('when sports are not provided', () => {
    const component = shallow(
      <SideMenu
        sports={null}
        sportItemClick={sportItemClick}
        currentSportId={currentSportId}
      />
    );
    it('should match the snapshot 2', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
