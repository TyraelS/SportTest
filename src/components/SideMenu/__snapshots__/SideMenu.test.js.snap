// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the SideMenu component when sports are not provided should match the snapshot 2 1`] = `
<ForwardRef(styled.div)>
  Smth went really wrong
</ForwardRef(styled.div)>
`;

exports[`Given the SideMenu component when the component is rendered should match the snapshot 1 1`] = `
<ForwardRef(styled.div)>
  <SideMenuItemContainer
    active={true}
    id="SBTC_01"
    key="SBTC_01"
    onClick={[MockFunction]}
  >
    SortofLeague
     - 
  </SideMenuItemContainer>
  <SideMenuItemContainer
    active={false}
    id="SBTC_02"
    key="SBTC_02"
    onClick={[MockFunction]}
  >
    AnotherLeague
     - 
  </SideMenuItemContainer>
</ForwardRef(styled.div)>
`;
